### PR TITLE
Signup error handling

### PR DESCRIPTION
**Pull request by Edward-Phillips**
added error handling for trying to sign up with the same email twice.

### CODE REVIEW

**Comment by FayeCarter** [comment]
LGTM

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -3,9 +3,14 @@ def new
...
   def create
-    @user = User.create(user_params)
-    session[:user_id] = @user.id
-    redirect_to posts_path
+    @user = User.new(user_params)
+    if @user.save 
+      session[:user_id] = @user.id
+      redirect_to posts_path
+    else
+      flash[:warning] = "That email is already in use"
```
Comment (tommyrharper) [inline app/controllers/users_controller.rb:11]: Good warning message :)

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -1,4 +1,7 @@
 <h1>Sign Up</h1>
+<% if flash[:warning] %>
+<h2> <%= flash[:warning] %></h2>
```
Comment (tommyrharper) [inline app/views/users/new.html.erb:3]: Nice, this will be totally obsolete now!

File: spec/features/user_can_singup_spec.rb [hunk 3]
```diff
@@ -13,4 +13,18 @@
...
+
+    fill_in "user[first_name]", with: "Kevin"
+    fill_in "user[last_name]", with: "Bacon"
+    fill_in "user[email]", with: "email@bacon.com"
+    fill_in "user[password]", with: "password"
+    click_button "Submit"
+    
+
+    expect(page).to have_content("That email is already in use")
+  end
```
Comment (tommyrharper) [inline spec/features/user_can_singup_spec.rb:29]: fantastic test :)

**Comment by tommyrharper** [comment]
Incredible code

**Pull request merged**